### PR TITLE
Support Spark-3.0.0 

### PR DESCRIPTION
**Pull request by HongW2019**
## What changes were proposed in this pull request?

Support Spark-3.0.0 for OAP

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,10 +11,12 @@ jobs:
...
-      - # Download spark 2.4.4
-      - "[ -f spark ] || mkdir spark && cd spark && wget http://archive.apache.org/dist/spark/spark-2.4.4/spark-2.4.4-bin-hadoop2.7.tgz && cd .."
-      - "tar -xf ./spark/spark-2.4.4-bin-hadoop2.7.tgz"
-      - "export SPARK_HOME=`pwd`/spark-2.4.4-bin-hadoop2.7"
+      - # Download spark 3.0.0
+      - "[ -f spark ] || mkdir spark && cd spark"
+      - "git clone -b rc1-add-hivethrift https://github.com/HongW2019/spark.git"
+      - "cd spark"
+      - "./build/mvn -DskipTests clean -Pyarn  -Phive-1.2 -Phive-thriftserver  install"
+      - "export SPARK_HOME=`pwd`/spark"
```
Comment (gczsjdy) [inline .travis.yml:19]: Glad to see this! Should we download and depend on this branch? https://github.com/Intel-bigdata/spark/tree/oap-master

Comment (gczsjdy) [reply .travis.yml:19]: cc @zhixingheyi-tian

Comment (HongW2019) [reply .travis.yml:19]: Yes, we have intended to, but for now, this Spark branch can not build successfully for it need  dependency --- OAP/oap-common in core/pom.xml, and we haven't uploaded it to Maven repo.

Comment (HongW2019) [reply .travis.yml:19]: Yes, we have intended to, but for now, this Spark branch can not build successfully,  for it needs dependency --- OAP/oap-common in core/pom.xml.

Comment (gczsjdy) [reply .travis.yml:19]: - Do we need to push OAP-common to maven central repo?
- I saw you used HongW2019/spark, why not Intel-bigdata/spark?

Comment (HongW2019) [reply .travis.yml:19]: Yes, we will fix oap-common dependency issue and  numa patch ASAP to help build Spark successfully,  then I will update branch to Intel-bigdata/spark in .travis.yml.

File: oap-cache/oap/src/test/scala/org/apache/spark/sql/execution/datasources/oap/CacheHotTablesSuite.scala [hunk 2]
```diff
@@ -199,140 +199,4 @@ class CacheHotTablesForParquetSuite extends CacheHotTablesSuite {
       )
     }
   }
-
-  test("Scan : Optimized") {
```
Comment (HongW2019) [inline oap-cache/oap/src/test/scala/org/apache/spark/sql/execution/datasources/oap/CacheHotTablesSuite.scala:203]: Spark-3.0 has dropped the `FROM-only` statement, so 'FROM parquet_test' from verifyScan is not allowed here, quick refer to apache/spark@3ddc77d

File: oap-cache/oap/src/test/scala/org/apache/spark/sql/execution/datasources/oap/CacheHotTablesSuite.scala [hunk 3]
```diff
@@ -199,140 +199,4 @@ class CacheHotTablesForParquetSuite extends CacheHotTablesSuite {
...
 }
-
-// class CacheHotTablesForOrcSuite extends CacheHotTablesSuite {
-//  protected def testTableName: String = "orc_test"
-//
-//  protected def fileFormat: String = "orc"
-//
-//  test("Project-> Filter -> Scan : Optimized") {
-//    withSQLConf(OapConf.OAP_ORC_DATA_CACHE_ENABLED.key -> "true",
-//      SQLConf.ORC_COPY_BATCH_TO_SPARK.key -> "true") {
```
Comment (HongW2019) [inline oap-cache/oap/src/test/scala/org/apache/spark/sql/execution/datasources/oap/CacheHotTablesSuite.scala]: SQLConf.ORC_COPY_BATCH_TO_SPARK has been dropped in Spark-3.0

File: oap-cache/oap/src/test/scala/org/apache/spark/sql/execution/datasources/oap/OapFileSourceStrategySuite.scala [hunk 4]
```diff
@@ -175,13 +175,6 @@ class OapFileSourceStrategyForParquetSuite extends OapFileSourceStrategySuite {
       (plan1, plan2) => plan1.sameResult(plan2)
     )
   }
-
-  test("Scan : Not Optimized") {
-    verifyScan(
```
Comment (HongW2019) [inline oap-cache/oap/src/test/scala/org/apache/spark/sql/execution/datasources/oap/OapFileSourceStrategySuite.scala:180]: Spark-3.0 has dropped the FROM-only statement, so 'FROM parquet_test' from verifyScan is not allowed here, quick refer to https://github.com/apache/spark/commit/3ddc77d9ac47968678ec43e28aa5efac9f2a91ad

**Pull request merged**